### ISSUE TITLE
Google -> TSV migration dashboard

### ISSUE THREAD

**Issue by turbomam** [GenomicsStandardsConsortium/mixs#522]
We need script(s) that can iteratively document the progress of converting the MIxS Google sheets into schemasheets TSVs, with no duplications or contradictions (except in the case of slot usages)

Ideally it should be easy for a non-developer to run

From 
- [**MIxS6 packages - Final_clean**](https://docs.google.com/spreadsheets/d/1QDeeUcDqXes69Y2RjU2aWgOpCVWo5OVsBX9MKmMqi_o/edit#gid=750683809)
- [MIxS6 Core- Final_clean](https://docs.google.com/spreadsheets/d/1QDeeUcDqXes69Y2RjU2aWgOpCVWo5OVsBX9MKmMqi_o/edit#gid=178015749)

to
- schemasheets/tsv_in/MIxS_6_term_updates_classdefs.tsv
- schemasheets/tsv_in/MIxS_6_term_updates_global_partial_slotdefs.tsv
- schemasheets/tsv_in/MIxS_6_term_updates_slot_assignments_and_usages.tsv

Questions:
- input? preferably the TSVs.
- output?

--- 

- [ ] Make sure that each class and term/slot is defined once and only once.
- [x] Every class mentioned in `schemasheets/tsv_in/MIxS_6_term_updates_slot_assignments_and_usages.tsv` should appear in `schemasheets/tsv_in/MIxS_6_term_updates_classdefs.tsv`
- [x] Every class mentioned in `schemasheets/tsv_in/MIxS_6_term_updates_classdefs.tsv` should appear in`schemasheets/tsv_in/MIxS_6_term_updates_slot_assignments_and_usages.tsv` 
- [x] Every term/slot mentioned in `schemasheets/tsv_in/MIxS_6_term_updates_slot_assignments_and_usages.tsv` should appear in `schemasheets/tsv_in/MIxS_6_term_updates_global_partial_slotdefs.tsv`
- [x] Every term/slot mentioned in `schemasheets/tsv_in/MIxS_6_term_updates_classdefs.tsv` should appear in`schemasheets/tsv_in/MIxS_6_term_updates_global_partial_slotdefs.tsv` 
- [x] we should identify the root cause of the multiple term definitions in [**MIxS6 packages - Final_clean**](https://docs.google.com/spreadsheets/d/1QDeeUcDqXes69Y2RjU2aWgOpCVWo5OVsBX9MKmMqi_o/edit#gid=750683809)

**Comment by turbomam** [comment]
One way to access these reports is though a web API with a Swagger interface. It's IP address isn't stable right now, but can be obtained from [Current NMDC utilities API address Google Doc](https://docs.google.com/document/d/1BKPFvsq6DOewSYrB6HI2lEppY-ZS-wHYYNEuFzujRyU/edit)

**Comment by ramonawalls** [comment]
> [] we should identify the root cause of the multiple term definitions in [MIxS6 packages - Final_clean](https://docs.google.com/spreadsheets/d/1QDeeUcDqXes69Y2RjU2aWgOpCVWo5OVsBX9MKmMqi_o/edit#gid=750683809)

I'm almost certain it is because we made this sheet manually. Unless @only1chunts disagrees, I think we can check that box.

**Comment by ramonawalls** [comment]
Assigning myself to complete the task of making sure each term or class has only one definition. I'll seek help as needed.

**Comment by ramonawalls** [comment]
> We need script(s) that can iteratively document the progress of converting the MIxS Google sheets into schemasheets TSVs, with no duplications or contradictions (except in the case of slot usages)

My understanding from our last call was that I would edit the TSV sheets so that we don't need to run the google sheet to TSV conversion ever again, right? However, I think this issue is about writing scripts to confirm that my manual edits to the TSVs accurately reflect (with corrections) the content of the google sheets. Am I correct in that understanding?

**Comment by only1chunts** [comment]
I agree with everything Ramona has added here, i.e.
 - definitions were a mistake caused by manual editing of sheets
 - moving forwards TSV sheets should be being used for editing, not G-Docs

Some form of documentation of changes being made between the released v6 and the new v7 (or 6.x) needs to be maintained, whether that is via scripts or manually done I dont mind. (As its a one-off process its probably easier to just be a manual thing maybe?). For future versions, it would be nice to have scripts that compare new to old to give a list of the changes, but presumably thats much easier when both versions are in the same format!

**Comment by sujaypatil96** [comment]
Not a concern anymore after transition to MIxS v6.2.
